io/net/virt-net/network_virtualization.py

```diff
@@ -420,6 +423,44 @@ def test_vnic_auto_failover(self):
         else:
             self.cancel("Provide more backing device, only 1 given")
 
+    def test_rmdev_viosfailover(self):
+        '''
+        using mrdev and mkdev command to check vios failover
+        '''
+        self.login(self.vios_ip, self.vios_user, self.vios_pwd)
+        cmd = "lsmap -all -vnic -cpid %s" % self.lpar_id
+        vnc_server = self.run_command(cmd)[2].split()[0]
+        cmd = "lsmap -vnic -vadapter %s" % vnc_server
+        output = self.run_command(cmd)
+        for line in output:
+            if 'Backing device' in line:
+                vnic_backing_device = line.split(':')[-1]
```

Comment:
```
else:
    self.fail
``` ?

Reply:
we can't fail here, their is so many lines in output.that will be fail always.

Reply:
@smruti77 ok, then we should run the `backing device remove` commands only when this `vnic_backing_device` exists.